app/src/main/java/org/mozilla/fenix/compose/button/Button.kt

```diff
@@ -31,55 +29,213 @@ import org.mozilla.fenix.theme.FirefoxTheme
             fontSize = 14.sp,
-            fontFamily = FontFamily(Font(R.font.metropolis_semibold)),
-            letterSpacing = 0.sp,
-            maxLines = 2
+            letterSpacing = 0.25.sp,
+            maxLines = 1
         )
     }
 }
 
+/**
+ * Primary button.
+ *
+ * @param text The button text to be displayed.
+ * @param icon Optional [Painter] used to display an [Icon] before the button text.
+ * @param onClick Invoked when the user clicks on the button.
+ */
+@Composable
+fun PrimaryButton(
+    text: String,
+    icon: Painter? = null,
+    onClick: () -> Unit,
+) {
+    Button(
+        text = text,
+        contentColor = FirefoxTheme.colors.textActionPrimary,
+        backgroundColor = FirefoxTheme.colors.actionPrimary,
+        icon = icon,
+        tint = FirefoxTheme.colors.iconActionPrimary,
+        onClick = onClick,
+    )
+}
+
+/**
+ * Secondary button.
+ *
+ * @param text The button text to be displayed.
+ * @param icon Optional [Painter] used to display an [Icon] before the button text.
+ * @param onClick Invoked when the user clicks on the button.
+ */
+@Composable
+fun SecondaryButton(
+    text: String,
+    icon: Painter? = null,
+    onClick: () -> Unit,
+) {
+    Button(
+        text = text,
+        contentColor = FirefoxTheme.colors.textActionSecondary,
+        backgroundColor = FirefoxTheme.colors.actionSecondary,
+        icon = icon,
+        tint = FirefoxTheme.colors.iconActionSecondary,
+        onClick = onClick,
+    )
+}
+
+/**
+ * Tertiary button.
+ *
+ * @param text The button text to be displayed.
+ * @param icon Optional [Painter] used to display an [Icon] before the button text.
+ * @param onClick Invoked when the user clicks on the button.
+ */
+@Composable
+fun TertiaryButton(
+    text: String,
+    icon: Painter? = null,
+    onClick: () -> Unit,
+) {
+    Button(
+        text = text,
+        contentColor = FirefoxTheme.colors.textActionTertiary,
+        backgroundColor = FirefoxTheme.colors.actionTertiary,
+        icon = icon,
+        tint = FirefoxTheme.colors.iconActionTertiary,
+        onClick = onClick,
+    )
+}
+
+/**
+ * Destructive button.
+ *
+ * @param text The button text to be displayed.
+ * @param icon Optional [Painter] used to display an [Icon] before the button text.
+ * @param onClick Invoked when the user clicks on the button.
+ */
+@Composable
+fun DestructiveButton(
+    text: String,
+    icon: Painter? = null,
+    onClick: () -> Unit,
+) {
+    Button(
+        text = text,
+        contentColor = FirefoxTheme.colors.textWarningButton,
+        backgroundColor = FirefoxTheme.colors.actionSecondary,
+        icon = icon,
+        tint = FirefoxTheme.colors.iconWarningButton,
+        onClick = onClick,
+    )
+}
+
 @Composable
 @Preview
-private fun PrimaryButtonPreview() {
+private fun ButtonLightThemePreview() {
     FirefoxTheme(theme = Theme.getTheme(isPrivate = false)) {
-        Box(Modifier.background(FirefoxTheme.colors.layer1)) {
```

Comment:
Alternatively, you can have one preview function but then do two `@Preview` calls:

`@Preview(uiMode = Configuration.UI_MODE_NIGHT_YES)`
`@Preview(uiMode = Configuration.UI_MODE_NIGHT_NO)`